### DISCUSSION TITLE
Alternate Tunneling Credentials

### DISCUSSION THREAD

**Discussion by dgitw** [devcontainers/community#49] [Q&A] [answered]
Currently tunnelling only allows Github credentials (obviously feature is geared towards Codespaces). What's the plan for alternate tunneling credentials?

Scenario: Currently we are users of DevContainers using remote VM's / SSH which has its downsides (replicating source to remote for one). We are using a self-hosted instance Github Enterprise Server leveraging Azure Active Directory credentials.

Is tunneling using AAD going to be supported in the future? Is tunneling seen as a mechanism for removing some of the friction from the Remote SSH / DevContainer scenario going forward?

**Comment by bamurtaugh** [comment, marked as answer]
@connor4312 @curib I thought we had an issue tracking this, but I can't seem to find it. Do you happen to know of one?

**Reply by curib** [reply]
@bamurtaugh, found it! Looks like it was closed. https://github.com/microsoft/vscode-remote-release/issues/8150

**Reply by bamurtaugh** [reply]
Ah thank you so much @curib! 

@dgitw it'd be great if we can track your request in the existing issue: https://github.com/microsoft/vscode-remote-release/issues/8150.

**Reply by dgitw** [reply]
@bamurtaugh  added - thanks.
